web/hooks/use-group-dragndrop.ts

```diff
@@ -0,0 +1,124 @@
+import { GroupDropLocation, handleGroupDragDrop } from "@/components/issues/issue-layouts/utils";
+import { ISSUE_FILTER_DEFAULT_DATA } from "@/store/issue/helpers/issue-helper.store";
+import { TIssue, TIssueGroupByOptions, TIssueOrderByOptions } from "@plane/types";
+import { TOAST_TYPE, setToast } from "@plane/ui";
+import { useIssueDetail, useIssues } from "./store";
+import { EIssuesStoreType } from "@/constants/issue";
+import { useRouter } from "next/router";
+import { useIssuesActions } from "./use-issues-actions";
+
+type DNDStoreType =
+  | EIssuesStoreType.PROJECT
+  | EIssuesStoreType.MODULE
+  | EIssuesStoreType.CYCLE
+  | EIssuesStoreType.PROJECT_VIEW
+  | EIssuesStoreType.DRAFT
+  | EIssuesStoreType.PROFILE
+  | EIssuesStoreType.ARCHIVED;
+
+export const useGroupIssuesDragNDrop = (
+  storeType: DNDStoreType,
+  orderBy: TIssueOrderByOptions | undefined,
+  groupBy: TIssueGroupByOptions | undefined,
+  subGroupBy?: TIssueGroupByOptions
+) => {
+  const router = useRouter();
+  const { workspaceSlug } = router.query;
+
+  const {
+    issue: { getIssueById },
+  } = useIssueDetail();
+  const { updateIssue } = useIssuesActions(storeType);
+  const {
+    issues: { getIssueIds },
+  } = useIssues(storeType);
+  const {
+    issues: { addCycleToIssue, removeCycleFromIssue },
+  } = useIssues(EIssuesStoreType.CYCLE);
```

Comment:
We would not need these methods from issue store if we store issue as a class, right?

Reply:
yes, we wont need to get it individually from cycles and modules issue store